[BE][FSDP] Subtest prefetching in `test_mixed_precision_e2e_full_shard()`

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81158
* __->__ #80915
* #80908
* #80873

This moves the forward and backward prefetching to be subtests just like the previous PR. This only targets the single test `test_mixed_precision_e2e_full_shard()`. However, that test originally corresponded to 161 out of 170 tests from the `test_fsdp_mixed_precision.py` file, so this covers the bulk of the TTS contribution.

**On the AI AWS cluster, this reduces the `test_mixed_precision_e2e_full_shard()` TTS from ~1200 seconds (20 minutes) to 270 seconds (4.5 minutes).**

Commit:
[BE][FSDP] Subtest prefetching in
`test_mixed_precision_e2e_full_shard()`

[ghstack-poisoned]

Commit:
Update on "[BE][FSDP] Subtest prefetching in `test_mixed_precision_e2e_full_shard()`"


This moves the forward and backward prefetching to be subtests just like the previous PR. This only targets the single test `test_mixed_precision_e2e_full_shard()`. However, that test originally corresponded to 161 out of 170 tests from the `test_fsdp_mixed_precision.py` file, so this covers the bulk of the TTS contribution.

**On the AI AWS cluster, this reduces the `test_mixed_precision_e2e_full_shard()` TTS from ~1200 seconds (20 minutes) to 270 seconds (4.5 minutes).**


[ghstack-poisoned]

Commit:
Update on "[BE][FSDP] Subtest prefetching in `test_mixed_precision_e2e_full_shard()`"


This moves the forward and backward prefetching to be subtests just like the previous PR. This only targets the single test `test_mixed_precision_e2e_full_shard()`. However, that test originally corresponded to 161 out of 170 tests from the `test_fsdp_mixed_precision.py` file, so this covers the bulk of the TTS contribution.

**On the AI AWS cluster, this reduces the `test_mixed_precision_e2e_full_shard()` TTS from ~1200 seconds (20 minutes) to 270 seconds (4.5 minutes).**


[ghstack-poisoned]

Commit:
Update on "[BE][FSDP] Subtest prefetching in `test_mixed_precision_e2e_full_shard()`"


This moves the forward and backward prefetching to be subtests just like the previous PR. This only targets the single test `test_mixed_precision_e2e_full_shard()`. However, that test originally corresponded to 161 out of 170 tests from the `test_fsdp_mixed_precision.py` file, so this covers the bulk of the TTS contribution.

**On the AI AWS cluster, this reduces the `test_mixed_precision_e2e_full_shard()` TTS from ~1200 seconds (20 minutes) to 270 seconds (4.5 minutes).**


[ghstack-poisoned]

Commit:
Update on "[BE][FSDP] Subtest prefetching in `test_mixed_precision_e2e_full_shard()`"


This moves the forward and backward prefetching to be subtests just like the previous PR. This only targets the single test `test_mixed_precision_e2e_full_shard()`. However, that test originally corresponded to 161 out of 170 tests from the `test_fsdp_mixed_precision.py` file, so this covers the bulk of the TTS contribution.

**On the AI AWS cluster, this reduces the `test_mixed_precision_e2e_full_shard()` TTS from ~1200 seconds (20 minutes) to 270 seconds (4.5 minutes).**


[ghstack-poisoned]

rohan-varma commented on test/distributed/fsdp/test_fsdp_mixed_precision.py:
```diff
@@ -86,40 +87,30 @@
-forward_prefetch_config = ["forward_prefetch", "no_forward_prefetch"]
 full_precision_param_dtype_config = [torch.float32, torch.float64]
-sharded_grad_scaler = ["enable_sharded_grad_scaler", None]
+enable_sharded_grad_scaler = ["enable_sharded_grad_scaler", None]
```
shall we make this [True, False] for simplicity?

rohan-varma reviewed (approved):
LGTM, thanks!

awgu commented:
@pytorchbot merge

Commit:
Update on "[BE][FSDP] Subtest prefetching in `test_mixed_precision_e2e_full_shard()`"


This moves the forward and backward prefetching to be subtests just like the previous PR. This only targets the single test `test_mixed_precision_e2e_full_shard()`. However, that test originally corresponded to 161 out of 170 tests from the `test_fsdp_mixed_precision.py` file, so this covers the bulk of the TTS contribution.

**On the AI AWS cluster, this reduces the `test_mixed_precision_e2e_full_shard()` TTS from ~1200 seconds (20 minutes) to 270 seconds (4.5 minutes).**


[ghstack-poisoned]

awgu commented:
@pytorchbot merge
